spring-integration-core/src/main/java/org/springframework/integration/mapping/AbstractHeaderMapper.java

```diff
@@ -308,7 +310,9 @@ private ResolvableType createJsonResolvableTypHeaderInAny(Object typeId, @Nullab
 			@Nullable Object keyId) {
 
 		try {
-			return JsonHeaders.buildResolvableType(getClassLoader(), typeId, contentId, keyId);
+			var classLoader = getClassLoader();
+			Assert.state(classLoader != null, "No ClassLoader available");
```

Comment:
Looks like Spring deals with null `ClassLoader` everywhere.
We should revise that `JsonHeaders.buildResolvableType()` to accept `null`, respectively.
In the end it uses `ClassUtils.forName()` which accepts null well.